monkey/monkey_island/cc/resources/pba_file_upload.py

```diff
@@ -77,15 +88,28 @@ def post(self, target_os):
         safe_filename = sanitize_filename(file_storage.filename)
 
         self._file_storage_service.save_file(safe_filename, file_storage.stream)
-        ConfigService.set_config_value(
-            (PBA_LINUX_FILENAME_PATH if target_os == LINUX_PBA_TYPE else PBA_WINDOWS_FILENAME_PATH),
-            safe_filename,
-        )
+        try:
+            self._update_config(target_os, safe_filename)
```

Comment:
This is probably a bit inconsistent with our configuration. As in, other config sections only get saved upon submit, this will get saved on file selection. It's not a big deal, but something to be aware of

Reply:
Agreed, also https://github.com/guardicore/monkey/pull/2087/files#r921027253